src/ejabberd_s2s_out.erl

```diff
@@ -55,35 +55,62 @@
          print_state/1,
          code_change/4]).
 
+-export_type([connection_info/0]).
+
 -ignore_xref([open_socket/2, print_state/1,
-              reopen_socket/2, start_link/3, stream_established/2,
+              reopen_socket/2, start_link/2, stream_established/2,
               wait_before_retry/2, wait_for_auth_result/2,
               wait_for_features/2, wait_for_starttls_proceed/2, wait_for_stream/2,
               wait_for_stream/2, wait_for_validation/2]).
 
+-type verify_requester() :: false | {S2SIn :: pid(), Key :: ejabberd_s2s:s2s_dialback_key(), SID :: ejabberd_s2s:stream_id()}.
+
 -include("mongoose.hrl").
 -include("jlib.hrl").
 
 -record(state, {socket,
-                streamid,
-                remote_streamid = <<>>,
-                use_v10,
+                streamid                :: ejabberd_s2s:stream_id() | undefined,
+                remote_streamid = <<>>  :: ejabberd_s2s:stream_id(),
+                use_v10                 :: boolean(),
                 tls = false             :: boolean(),
                 tls_required = false    :: boolean(),
                 tls_enabled = false     :: boolean(),
                 tls_options             :: mongoose_tls:options(),
                 authenticated = false   :: boolean(),
-                db_enabled = true       :: boolean(),
+                dialback_enabled = true :: boolean(),
                 try_auth = true         :: boolean(),
-                myname, server, queue,
+                from_to                 :: ejabberd_s2s:fromto(),
+                myname                  :: jid:lserver(),
+                server                  :: jid:lserver(),
+                queue                   :: element_queue(),
                 host_type               :: mongooseim:host_type(),
-                delay_to_retry = undefined_delay,
-                new = false             :: boolean(),
-                verify = false          :: false | {pid(), Key :: binary(), SID :: binary()},
+                delay_to_retry = undefined :: undefined | non_neg_integer(),
```

Comment:
Minor: `undefined` is the default value, so there is no need to explicitly assign it here.